pysparkling/sql/expressions/mappers.py

```diff
@@ -75,6 +87,115 @@ def __str__(self):
         return "({0} IS NOT NULL)".format(self.column)
 
 
+class Cast(Expression):
+    def __init__(self, e, dtype):
+        super().__init__(e, dtype)
+
+        self.e = e
+        if not isinstance(dtype, str):
+            try:
+                dtype = dtype.typeName()
+            except AttributeError:
+                pass
+        self.dtype = dtype
+        self.dtype_str = dtype
+        if self.dtype_str == 'string':
+            self.dtype = str
+        elif self.dtype_str == 'integer':
+            self.dtype = int
+        elif self.dtype_str == 'boolean':
+            self.dtype = bool
+        elif self.dtype_str in ['str', 'int', 'float', 'bool']:
+            self.dtype = eval(self.dtype_str)
```

Comment:
I'm not a huge fan of eval, would you mind doing something without it?

For instance something like

```python
mapping = {
    'string': str,
    'str': str,
    'integer': int,
    'int': int,
    'boolean': bool,
    'bool': bool,
    'float': float
}

try:
    self.dtype = mapping[dtype_str]
except KeyError:
    raise NotImplementedError("Unknown cast type: {}".format(dtype))
```

Reply:
"str" does not seem supported by spark btw, only "string"

Reply:
Yeah that works. I'll fix this one.